### PR TITLE
Autocomplete changes related to mobile modals PR

### PR DESCRIPTION
**Pull request by joel-jeremy**
Split changes from #2472 for easier review.

### CODE REVIEW

File: packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx [hunk 1]
```diff
@@ -159,7 +168,7 @@ export function CategoryAutocomplete({
   );
 
   return (
-    <Autocomplete
+    <Autocomplete<CategoryAutocompleteItem>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx:171]: **❓ question:** ‏do we need this? Previously the type could be inferred from  `items`

File: packages/desktop-client/src/components/autocomplete/AccountAutocomplete.tsx [hunk 2]
```diff
@@ -180,17 +218,20 @@ export function AccountItem({
...
           borderRadius: embedded ? 4 : 0,
+          ...narrowStyle,
         },
       ])}`}
       data-testid={`${item.name}-account-item`}
       data-highlighted={highlighted || undefined}
       {...props}
     >
-      {item.name}
+      <TextOneLine>{item.name}</TextOneLine>
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/AccountAutocomplete.tsx:228]: **👏 praise:** ‏good idea

File: packages/desktop-client/src/components/autocomplete/Autocomplete.tsx [hunk 3]
```diff
@@ -41,7 +74,7 @@ function findItem<T extends Item>(
   return value;
 }
 
-function getItemName(item: null | string | Item): string {
+function getItemName<T extends Item>(item: T | T['name']): string {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.tsx:77]: **🔨 warning:** ‏`item` can also be `null`
```suggestion
function getItemName<T extends Item>(item: T | T['name'] | null): string {
```

File: packages/desktop-client/src/components/autocomplete/Autocomplete.tsx [hunk 4]
```diff
@@ -239,6 +247,26 @@ function SingleAutocomplete<T extends Item>({
...
+      ...narrowInputStyle,
+      ...inputProps.style,
+    },
+  };
+
+  useEffect(() => {
+    if (!isOpen) {
+      onClose?.();
+    }
+  }, [isOpen, onClose]);
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.tsx:269]: **❓ question:** ‏is `onClose` called if the component is mounted with `isOpen = false`? Is that correct? Could it lead to some issues?

File: packages/desktop-client/src/components/autocomplete/Autocomplete.tsx [hunk 5]
```diff
@@ -579,36 +607,38 @@ function MultiItem({ name, onRemove }: MultiItemProps) {
...
+
 function MultiAutocomplete<T extends Item>({
   value: selectedItems,
   onSelect,
   suggestions,
   strict,
+  clearOnBlur = true,
   ...props
 }: MultiAutocompleteProps<T>) {
+  selectedItems = selectedItems || [];
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.tsx:624]: **💬 suggestion:** ‏ideally the props should be treated as immutable. Could we move the default value to line 617?

```
value: selectedItems = [],
```

File: packages/desktop-client/src/components/autocomplete/Autocomplete.tsx [hunk 6]
```diff
@@ -626,10 +656,12 @@ function MultiAutocomplete<T extends Item>({
   return (
     <Autocomplete
       {...props}
-      isMulti
+      type="single"
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.tsx:659]: **🔨 warning:** ‏this seems odd. Shouldn't it be `multi`?

```
type="multi"
```

File: packages/desktop-client/src/components/transactions/TransactionsTable.jsx [hunk 7]
```diff
@@ -521,12 +521,11 @@ function PayeeCell({
...
             }}
             showManagePayees={true}
-            tableBehavior={true}
+            clearOnBlur={false}
             focused={true}
-            onUpdate={onUpdate}
+            onUpdate={(id, value) => onUpdate?.(value)}
             onSelect={onSave}
             onManagePayees={() => onManagePayees(payeeId)}
-            isCreatable
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/transactions/TransactionsTable.jsx:529]: **🔨 warning:** ‏same problem here. Would this lead to a regression?

**Review by MatissJanis — CHANGES_REQUESTED** [review]

File: packages/desktop-client/src/components/autocomplete/Autocomplete.tsx (hunk 6 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/autocomplete/Autocomplete.tsx:662]: This is correct, The `MultiAutocomplete` component uses the `SingleAutocomplete` internally which clears the selection on every select. I've renamed the `isMulti` to `clearOnSelect`.

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**